scivision/io/reader.py

```diff
@@ -81,3 +87,28 @@ def load_pretrained_model(
     install_package(config, allow_install=allow_install)
 
     return PretrainedModel(config)
+
+
+def load_dataset(
+    path: os.PathLike,
+    branch: str = "main"
+) -> intake.catalog.local.YAMLFileCatalog:
+    """Load a pre-trained model.
+
+    Parameters
+    ----------
+    path : PathLike
+        The filename, path or URL of an intake catalog, which links to a dataset.
+    branch : str, default = main
+        Specify the name of a github branch if loading from github.
+
+    Returns
+    -------
+    intake.catalog.local.YAMLFileCatalog
+        The intake catalog object from which an xarray dataset can be created.
+    """
+    config = _parse_config(path, branch=branch, model_config=False)
+    tempdir = tempfile.mkdtemp()
+    with open(tempdir + '/scivision.yml', 'w') as file:
+        yaml.dump(config, file)
```

Comment:
Can intake use a bytestream to prevent having to write the file to disk?

Reply:
Rather than loading the config at all, could we hand the path straight to `intake.open_catalog`?